docs/guidelines.md

```diff
@@ -0,0 +1,87 @@
+# Feedback guidelines
+
+The goal of a language track on Exercism is to give people a way to achieve a
+high level of fluency at a low level of proficiency. We're aiming for fluency
+in the syntax, idioms, and the standard library of the language. You can read
+more about the [goal of exercism here](https://github.com/exercism/docs/blob/master/about/goal-of-exercism.md).
+
+## Definitions
+
+In the following paragraphs, keywords such as **MUST**, **SHOULD**, **COULD**
+are to be interpreted as in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt);
+given that we recognise the following four output states and their restrictions:
+
+- `approve_as_optimal`: **COULD** be a best solution, **SHOULD** be without comment
```

Comment:
Isn't `having a best solution` a requirement or at least a recommendation of `approve_as_optimal`?

Reply:
If `approve_as_optimal` can have a comment, what is the difference to `approve_with_comment`?

Reply:
`approve_as_optimal` is for solutions where no comment about an improved is required.

Reply:
So @tehsphinx if you take @iHiD comment and the one that was made about _random tips_, we might want to later use `approve_as_optimal` still but also add a comment.

The reason I used `COULD` and `SHOULD` here is that if I used `MUST`, technically a test framework could blow up the analyzer if a tip is added, or if the approved solution is not technically a "best" solution.

Reply:
@SleeplessByte I understand the reasoning, but to me `**COULD** be a best solution` still feels a bit odd. I would at least have expected **SHOULD** here, but maybe that's me?

Reply:
I'm indifferent. I'll change it so it's clearer!

Reply:
Done!

Reply:
Nice!

Reply:
@SleeplessByte I'm not seeing the change though :)